### PR TITLE
Dice score

### PR DESCRIPTION
**Pull request by myychal**
Quick pull with dice score. Will come handy during training.

### CODE REVIEW

File: tests/test_metrics.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        images = torch.ones(*BATCH_DIMS)
+        target = torch.ones(*BATCH_DIMS)
+        assert np.isclose(dice_score_one_class(images, target), 1, atol=1.e-4)
+
+    def test_if_returns_0_for_worst_fit(self):
+        images = torch.zeros(*BATCH_DIMS)
+        target = torch.ones(*BATCH_DIMS)
+        assert np.isclose(dice_score_one_class(images, target), 0, atol=1.e-4)
+
+    def test_if_works_for_non_binary_data(self):
```
Comment (khrynczenko) [inline tests/test_metrics.py]: Name is misleading. It should say test_assertion_error_for_non_binary_data or something like that.

File: brats/metrics.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import torch
+
+ZERO_AND_ONE = torch.tensor([0., 1.])
+CHANNELS_DIM = 1
+ONE_CLASS = 1
+
+
+def dice_score_one_class(prediction: torch.Tensor, target: torch.Tensor,
```
Comment (szympie) [inline brats/metrics.py]: There is a large amount of code overlapping with dice loss. Is it possible to extract it somehow and use in both loss and metric?

**Comment by myychal** [comment]
After our discussion, I have removed the assertion that the input and target need to be binary.

File: brats/metrics.py (hunk 2 above)
Comment (myychal) [reply brats/metrics.py]: Addressed. Extracted intersection and union calculations to separate functions and utilized dice **score** calculation in the dice **loss**

File: tests/test_metrics.py (hunk 1 above)
Comment (myychal) [reply tests/test_metrics.py]: Removed this test as we agreed that the input does not to be binary
